DATAREPO_CLIENT_README.md

```diff
@@ -0,0 +1,34 @@
+# Terra Data Repo Client project
+The tdrclient directory within the data repo code base contains an independent
+gradle project.
+
+## Build Targets
+The project has two targets that are interesting to use.
+
+### build
+The build target generates the OpenAPI client code for using the swagger source from the
+data repo code base. The code is generated into the `src/` directory. That allows the
+normal gradle java plugin to easily run the compile, build and test.
+
+### artifactoryPublish
```

Comment:
It isn't clear to me how the `artifactoryPublish` target actually ultimately executes the `build.gradle` below. Is it that the `build.gradle` below is executed serially line-by-line?

Reply:
artifactoryPublish is a gradle target, like `build` or `test`.
It (magically) has a dependency on the build target so if you just say:
```
../gradlew artifactoryPublish
```
It will perform the build step and then perform the publish step. The check at the top of the build.gradle file makes sure the envvars are defined before trying to do anything.

Does that make sense?

Reply:
Yes. The key line is [L25](https://github.com/DataBiosphere/jade-data-repo/pull/455/files#diff-8cc478a9cb5bcc9ecac8431e07912213R25), which hooks in the magic of the Artifactory plugin.